contracts/StakeManager.sol

```diff
@@ -251,6 +252,12 @@ contract StakeManager is Ownable {
         external
         onlyOldManager
     {
+        if (address(migration) != address(0)) {
+            revert StakeManager__PendingMigration();
+        }
+        if (currentEpoch > 0) {
+            revert StakeManager__AlreadyProcessedEpochs();
```

Comment:
This changed @3esmit I need to be get carefully reviewed by you.
There's an assumption here that: When a `StakeManager` migrates to a new one, that new one should not have processed any epochs yet.

The reason I've introduced this change is so that its currently defined invariants don't fail. Namely:

- `epochsOnlyIncrease` (this one fails when `migrationInitialize()` is called with arbitrary values and
- `highEpochsAreNull` (can also file when `migrationInitialize()` is called with arbitrary values

Obviously, this prevents `StakeManager` from migrating to a new one if the new one already started processing accounts itself.

I guess this can happen in practice, when we put out a second `StakeManager`, and before we initialize the migration, some account starts staking for example.

At this point, I'm even wondering if it makes sense to introduce some `migrationFinalized` flag and use that instead. Something for us to say: "this stakemanager is meant to be migrated to, please don't stake on this one yet"

We could use that flag instead of a `currentEpoch > 0` check.

Lemme know what you think.

Reply:
This certainly should be further discussed. 
I think is valid to think of, if the contract is constructed with migration, meaning it expects a coming migration, it should first wait the migration start to accept processing the epochs and accounts.